components/scream/src/physics/shoc/shoc_functions.hpp

```diff
@@ -68,6 +69,112 @@ struct Functions
 
   using Workspace = typename ekat::WorkspaceManager<Spack, Device>::Workspace;
 
+  // This struct stores input views for shoc_main.
+  struct SHOCInput {
+    // Grid spacing of host model in x direction [m]
+    view_1d<Pack1d> host_dx;
+    // grid spacing of host model in y direction [m]
+    view_1d<Pack1d> host_dy;
+    // heights, for thermo grid [m]
+    view_2d<Spack>  zt_grid;
+    // heights, for interface grid [m]
+    view_2d<Spack>  zi_grid;
+    // pressure levels on thermo grid [Pa]
+    view_2d<Spack>  pres;
+    // pressure levels on interface grid [Pa]
+    view_2d<Spack>  presi;
+    // Differences in pressure levels [Pa]
+    view_2d<Spack>  pdel;
+    // virtual potential temperature [K]
+    view_2d<Spack>  thv;
+    // large scale vertical velocity [m/s]
+    view_2d<Spack>  w_field;
+    // Surface sensible heat flux [K m/s]
+    view_1d<Pack1d> wthl_sfc;
+    // Surface latent heat flux [kg/kg m/s]
+    view_1d<Pack1d> wqw_sfc;
+    // Surface momentum flux (u-direction) [m2/s2]
+    view_1d<Pack1d> uw_sfc;
+    // Surface momentum flux (v-direction) [m2/s2]
+    view_1d<Pack1d> vw_sfc;
+    // Surface flux for tracers [varies]
+    view_2d<Spack>  wtracer_sfc;
+    // Exner function [-]
+    view_2d<Spack>  exner;
+    // Host model surface geopotential height
```

Comment:
Do we know the units for this?

Reply:
[m]

Reply:
Everywhere in shoc.F90 its given the same description. Others will know better than me though.. @bogensch